Swiften/MIX/MIX.h

```diff
@@ -59,11 +61,22 @@ namespace Swift {
             */
             virtual void updatePreferences(Form::ref form) = 0;
 
+            /**
+            * Register nick with MIX Channel.
+            */
+            virtual void registerNick(const std::string &nick) = 0;
```

Comment:
I'd think it would make sense to have `Swiften::MIX` deal with that and tell its user what features are supported. Probably makes sense to do that early in the beginning.